client/src/endpoint/account.rs

```diff
@@ -452,3 +471,175 @@ mod effects_tests {
+    /// # // Not making requests seeing as the main documentation already does this.
+    /// # // This serves to document the usage while conserving hits to horizon.
+    /// let endpoint = account::Payments::new("abc123").cursor("cursor");
+    /// ```
+    pub fn cursor(mut self, cursor: &str) -> Self {
+        self.cursor = Some(cursor.to_string());
+        self
+    }
+
+    /// Sets the maximum number of records to return.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// use stellar_client::endpoint::account;
+    /// # // Not making requests seeing as the main documentation already does this.
+    /// # // This serves to document the usage while conserving hits to horizon.
+    /// let endpoint = account::Payments::new("abc123").limit(20);
+    /// ```
+    pub fn limit(mut self, limit: u32) -> Self {
+        self.limit = Some(limit);
+        self
+    }
+
+    /// Fetches all records in a set order, either ascending or descending.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// use stellar_client::endpoint::{account, Order};
+    /// # // Not making requests seeing as the main documentation already does this.
+    /// # // This serves to document the usage while conserving hits to horizon.
+    /// let endpoint = account::Payments::new("abc123").order(Order::Asc);
+    /// ```
+    pub fn order(mut self, order: Order) -> Self {
+        self.order = Some(order);
+        self
+    }
+
+    fn has_query(&self) -> bool {
+        self.order.is_some() || self.cursor.is_some() || self.limit.is_some()
+    }
+}
+
+impl Cursor<Operation> for Payments {
+    fn cursor(self, cursor: &str) -> Self {
+        self.cursor(cursor)
+    }
+}
+
+impl IntoRequest for Payments {
+    type Response = Records<Operation>;
+
+    fn into_request(self, host: &str) -> Result<Request<Body>> {
+        let mut uri = format!("{}/accounts/{}/payments", host, self.id);
+        if self.has_query() {
+            uri.push_str("?");
+
+            if let Some(cursor) = self.cursor {
+                uri.push_str(&format!("cursor={}&", cursor));
+            }
+
+            if let Some(order) = self.order {
+                uri.push_str(&format!("order={}&", order.to_param()));
+            }
+
+            if let Some(limit) = self.limit {
+                uri.push_str(&format!("limit={}", limit));
+            }
+        }
+
+        let uri = Uri::from_str(&uri)?;
+        let request = Request::get(uri).body(Body::None)?;
+        Ok(request)
+    }
+}
+
+#[cfg(test)]
+mod payments_tests {
+    use super::*;
+
+    #[test]
+    fn it_leaves_off_the_params_if_not_specified() {
+        let payments = Payments::new("abc123");
+        let req = payments
+            .into_request("https://horizon-testnet.stellar.org")
+            .unwrap();
+        assert_eq!(req.uri().path(), "/accounts/abc123/payments");
+        assert_eq!(req.uri().query(), None);
+    }
+
+    #[test]
+    fn it_can_make_a_payments_uri() {
```

Comment:
Is there any significant difference between this and the above test?

Reply:
Nope. Merged the minor differences between the two.